src/concepts/plugins/frontend.md

```diff
@@ -0,0 +1,479 @@
+            "id",
+            "kind"
+          ],
+          "properties": {
+            "backend": {
+              "anyOf": [
+                {
+                  "$ref": "#/definitions/ManifestFrontendPluginConnection"
+                },
+                {
+                  "type": "null"
+                }
+              ]
+            },
+            "entrypoint": {
+              "type": [
+                "string",
+                "null"
+              ]
+            },
+            "id": {
+              "$ref": "#/definitions/ManifestID"
+            },
+            "kind": {
+              "type": "string",
+              "enum": [
+                "frontend"
+              ]
+            },
+            "name": {
+              "type": [
+                "string",
+                "null"
+              ]
+            },
+            "style": {
+              "type": [
+                "string",
+                "null"
+              ]
+            }
+          }
+        },
+        {
+          "type": "object",
+          "required": [
+            "entrypoint",
+            "id",
+            "kind",
+            "runtime"
+          ],
+          "properties": {
+            "entrypoint": {
+              "type": "string"
+            },
+            "id": {
+              "$ref": "#/definitions/ManifestID"
+            },
+            "kind": {
+              "type": "string",
+              "enum": [
+                "backend"
+              ]
+            },
+            "name": {
+              "type": [
+                "string",
+                "null"
+              ]
+            },
+            "runtime": {
+              "$ref": "#/definitions/ManifestBackendPluginRuntime"
+            }
+          }
+        }
+      ]
+    }
+  }
+}
+```
+
+</details>
+
+<details>
+<summary>TypeScript format.</summary>
+
+```ts
+export type Schema = Manifest
+export type ManifestID = string
+export type ManifestPlugin =
+  | {
+      backend?: ManifestFrontendPluginConnection | null
+      entrypoint: string
+      id: ManifestID
+      kind: "frontend"
+      name?: string | null
+      style: string
+      [k: string]: unknown
+    }
+  | {
+      entrypoint: string
+      id: ManifestID
+      kind: "backend"
+      name?: string | null
+      runtime: ManifestBackendPluginRuntime
+      [k: string]: unknown
+    }
+export type ManifestBackendPluginRuntime = "javascript"
+
+export interface Manifest {
+  author?: ManifestAuthor | null
+  description?: string | null
+  id: ManifestID
+  name?: string | null
+  plugins: ManifestPlugin[]
+  version: string
+  [k: string]: unknown
+}
+export interface ManifestAuthor {
+  email?: string | null
+  name?: string | null
+  url?: string | null
+  [k: string]: unknown
+}
+export interface ManifestFrontendPluginConnection {
+  id: ManifestID
+  [k: string]: unknown
+}
+```
+
+</details>
+:::
+
+### index.ts
+
+The example entrypoint file in the frontend starterkit is as follows:
```

Comment:
An example is not a bad thing, but in general in concept documentation you want to present small snippets of code at a time instead of one big blob.
Start with the init function, then build the example slowly by adding each functionality per paragraph.
I would suggest you look how programming books present their examples that would help you structure it.

Reply:
Lets change strategy, instead of explaining the full starterkit it would be better to craft the example toward it by doing incremental improvements to it until we reach something similar to the starterkit.